### PR TITLE
[jvm-packages] shade xgboost spark packages

### PR DESCRIPTION
**Pull request by wbo4958**
This PR assembles xgboost4j.jar/xgboost4j-spark.jar into the existing xgboost4j-spark.jar and xgboost4j.jar/xgboost4j-spark.jar/xgboost4j-spark-gpu.jar into xgboost4j-spark-gpu.jar. So with this PR, user can use xgboost4j-spark.jar for xgboost cases without specifying the xgboost4j.jar explicitly. Similary to xgboost4j-spark-gpu.jar

### CODE REVIEW

**Comment by wbo4958** [comment]
I did the tests manually on the local standalone cluster for both scala 2.12 and scala 2.13. All of the tests pass.

**Comment by wbo4958** [comment]
Hi @eordentlich, @trivialfis, Could you help review this PR, thx.

**Comment by trivialfis** [comment]
I will leave the review to @eordentlich or @hcho3 .

File: jvm-packages/xgboost4j-spark-gpu/pom.xml [hunk 1]
```diff
@@ -19,6 +19,27 @@
                     <skipAssembly>false</skipAssembly>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <configuration>
+                  <shadedArtifactAttached>false</shadedArtifactAttached>
+                  <createDependencyReducedPom>true</createDependencyReducedPom>
```
Comment (eordentlich) [inline jvm-packages/xgboost4j-spark-gpu/pom.xml:27]: Isn't this true by default?

File: jvm-packages/xgboost4j-spark/pom.xml [hunk 2]
```diff
@@ -19,6 +19,27 @@
                     <skipAssembly>false</skipAssembly>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <configuration>
+                  <shadedArtifactAttached>false</shadedArtifactAttached>
+                  <createDependencyReducedPom>true</createDependencyReducedPom>
```
Comment (eordentlich) [inline jvm-packages/xgboost4j-spark/pom.xml:27]: Same here.

**Review by eordentlich — COMMENTED** [review]
Looks like documentation needs to be updated to match these changes:
https://xgboost.readthedocs.io/en/latest/jvm/xgboost_spark_migration.html#xgboost-spark-packages
https://xgboost.readthedocs.io/en/latest/install.html#jvm

File: jvm-packages/xgboost4j-spark/pom.xml (hunk 2 above)
Comment (wbo4958) [reply jvm-packages/xgboost4j-spark/pom.xml]: yes, you're correct. 

fixed it.

File: jvm-packages/xgboost4j-spark-gpu/pom.xml (hunk 1 above)
Comment (wbo4958) [reply jvm-packages/xgboost4j-spark-gpu/pom.xml]: yes, you're correct. 

fixed it.

**Comment by wbo4958** [comment]
> Looks like documentation needs to be updated to match these changes: https://xgboost.readthedocs.io/en/latest/jvm/xgboost_spark_migration.html#xgboost-spark-packages https://xgboost.readthedocs.io/en/latest/install.html#jvm

I will have a follow up PR to update these completely.

**Comment by trivialfis** [comment]
> I will have a follow up PR to update these completely.

Could you please open a tracking issue?

**Comment by wbo4958** [comment]
Done. https://github.com/dmlc/xgboost/issues/10841

**Review by hcho3 — APPROVED** [review]

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**